### ISSUE TITLE
Php 7.4 is end of life and nextcloud instructions still use it

### ISSUE THREAD

**Issue by benmordecai** [LukeSmithxyz/landchad#254]
Nextcloud docs recommend php 8.1.

https://docs.nextcloud.com/server/latest/admin_manual/installation/system_requirements.html

As far as I can tell it should be as simple as changine 7.4 everywhere to 8.1

**Comment by LukeSmithxyz** [comment]
No, it doesn't look like it's in Debian yet, so this would at least break the installation or make it more difficult. When it's in the Debian repositories, we'll switch it up.

**Comment by benmordecai** [comment]
Great point, I forgot that I had to add a ppa to get the latest PHP.

**Comment by benmordecai** [comment]
In the process of completing my upgrade to PHP 8.1 I ran into two issues that took a while for me to trace out. The first is that I was getting tons of errors in `/var/log/nginx` because the config file presents both the use of a socket and a network port, so it was attempting to pass php-fpm data on both, but php-fpm was configured for the socket only. 

```
upstream php-handler {
    server unix:/var/run/php/php7.4-fpm.sock;
    server 127.0.0.1:9000; #comment this line out
}
```
AFAIK there is only one needed and the socket is preferable for security reasons. 

The second issue I ran into was I would occasionally get 502 Bad Gateway errors on nginx that would come and go randomly. This seemed to disappear once I set some environment variables in php-fpm. These are commented out by default but putting them in appears to help. I think this is due to the difference between when php-cli runs where it has these paths by default vs. php-fpm where it needs to be in the config.

`/etc/php/8.1/fpm/pool.d/www.conf`

```
env[HOSTNAME] = $HOSTNAME
env[PATH] = /usr/local/bin:/usr/bin:/bin
env[TMP] = /tmp
env[TMPDIR] = /tmp
env[TEMP] = /tmp
```

This is mentioned in the admininstration console at nextcloud.example.com/nextcloud/settings/admin/overview.
